File: app/src/main/java/co/tpcreative/supersafe/ui/main_tab/MainTabPresenter.java
package co.tpcreative.supersafe.ui.main_tab;
import android.util.Log;
import com.google.gson.Gson;
import co.tpcreative.supersafe.common.presenter.BaseView;
import co.tpcreative.supersafe.common.presenter.Presenter;
import co.tpcreative.supersafe.model.User;

public class MainTabPresenter extends Presenter<BaseView> {
    private static final String TAG = MainTabPresenter.class.getSimpleName();
    protected User mUser ;
    public MainTabPresenter(){

    }
    public void onGetUserInfo(){
        mUser = User.getInstance().getUserInfo();
    }

}

File: app/src/main/java/co/tpcreative/supersafe/ui/main_tab/MainTabPresenter.kt
package co.tpcreative.supersafe.ui.main_tab
import co.tpcreative.supersafe.common.presenter.BaseView
import co.tpcreative.supersafe.common.presenter.Presenter
import co.tpcreative.supersafe.common.util.Utils
import co.tpcreative.supersafe.model.EmptyModel
import co.tpcreative.supersafe.model.User

class MainTabPresenter : Presenter<BaseView<EmptyModel>>() {
    var mUser: User? = null
    fun onGetUserInfo() {
        mUser = Utils.getUserInfo()
    }
    companion object {
        private val TAG = MainTabPresenter::class.java.simpleName
    }
}